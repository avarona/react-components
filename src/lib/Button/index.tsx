import * as React from "react";
import classnames from "classnames";
import styles from "./styles.module.css";

type Props = {
  addClass?: string;
  children?: any;
  onClick?: () => void;
  type?: "primary" | "secondary" | "danger";
  disabled?: boolean;
};

const Button = ({
  addClass,
  children,
  onClick,
  type = "primary",
  disabled = false
}: Props) => (
  <button
    className={classnames(addClass, styles.button, styles[`button-${type}`], {
      [styles["button-disabled"]]: !!disabled
    })}
    onClick={onClick}
    disabled={disabled}
  >
    {children}
  </button>
);

export default Button;
