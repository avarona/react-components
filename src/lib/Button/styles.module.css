.button {
  text-align: center;
  outline: none;
  border-radius: 4px;
  padding: 8px;
  font-size: 12px;
  font-weight: 500;
  cursor: pointer;
}

.button-primary {
  background: blue;
  color: white;
}

.button-secondary {
  background: green;
  color: white;
}

.button-danger {
  background: red;
  color: white;
}

.button-disabled {
  background: grey;
  cursor: not-allowed;
}
