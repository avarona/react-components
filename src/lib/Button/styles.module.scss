.button {
  text-align: center;
  outline: none;
  border-radius: 4px;
  padding: 8px;
  font-size: 12px;
  font-weight: 500;
  cursor: pointer;

  &-primary {
    background: blue;
    color: white;
  }

  &-secondary {
    background: green;
    color: white;
  }

  &-danger {
    background: red;
    color: white;
  }

  &-disabled {
    background: grey;
    cursor: not-allowed;
  }
}
