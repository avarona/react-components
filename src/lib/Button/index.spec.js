import React from "react";
import { shallow } from "enzyme";

import Button from "./";

describe("Button Component", () => {
  it("renders", () => {
    const wrapper = shallow(<Button>Click Me!</Button>);

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find("button").exists()).toBe(true);
  });
});
