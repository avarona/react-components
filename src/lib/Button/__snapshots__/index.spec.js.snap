// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders 1`] = `"<button class=\\"button button-primary\\">Click Me!</button>"`;
